src/TableFunctions/TableFunctionFile.cpp

```diff
@@ -1,16 +1,73 @@
 #include <TableFunctions/TableFunctionFile.h>
 #include <TableFunctions/parseColumnsListForTableFunction.h>
 
+#include "Parsers/IAST_fwd.h"
 #include "registerTableFunctions.h"
 #include <Access/Common/AccessFlags.h>
 #include <Interpreters/Context.h>
 #include <Storages/ColumnsDescription.h>
 #include <Storages/StorageFile.h>
 #include <TableFunctions/TableFunctionFactory.h>
+#include <Interpreters/evaluateConstantExpression.h>
+#include <Formats/FormatFactory.h>
+#include <Parsers/ASTIdentifier_fwd.h>
 
 namespace DB
 {
 
+namespace ErrorCodes
+{
+    extern const int UNKNOWN_IDENTIFIER;
+    extern const int LOGICAL_ERROR;
+    extern const int BAD_ARGUMENTS;
+}
+
+void TableFunctionFile::parseFirstArguments(const ASTPtr & arg, ContextPtr context)
+{
+    if (context->getApplicationType() != Context::ApplicationType::LOCAL)
+    {
+        ITableFunctionFileLike::parseFirstArguments(arg, context);
+        return;
+    }
+
+    if (auto opt_name = tryGetIdentifierName(arg))
```

Comment:
Why don't we evaluate identifiers as literals for file table function with file descriptors? It's supposed that such queries should work:
`clickhouse local -q "select * from file(data.jsonl)"`
But now it throws an exception `Unknown identifier 'data.jsonl' in first arguments.`

Reply:
> Why don't we evaluate identifiers as literals for file table function with file descriptors? It's supposed that such queries should work: `clickhouse local -q "select * from file(data.jsonl)"` But now it throws an exception `Unknown identifier 'data.jsonl' in first arguments.` All identifiers except`stdin/stdout/stderr` should be interpreted as file names

okay, thank you !